### PR TITLE
Add storybook test github action

### PR DESCRIPTION
**Pull request by Gkrumbach07**
closes: https://github.com/opendatahub-io/odh-dashboard/issues/984
## Description

This adds storybook ci testing as a github action. The action will build storybook, run the build, run the test-runner, and if all passes, will pass the ci

## How Has This Been Tested?

1. fork the repo
2. make a not breaking pr
3. check that the storybook ci runs asn passes
4. make another pr with a change that breaks the test 
5. check that storybook ci runs and fails

## Test Impact

im not going to test the test

## Request review criteria:

- [x] The commits have meaningful messages (squashes happen on merge by the bot).
- [x] Included any necessary screenshots or gifs if it was a UI change.
- [x] Included tags to the UX team if it was a UI/UX change.
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has manually tested the changes and verified that the changes work
- [x] The developer has added tests or explained why testing cannot be added (unit tests & storybook for related changes)

### CODE REVIEW

**Review by lucferbux — CHANGES_REQUESTED** [review]

**Review by lucferbux — CHANGES_REQUESTED** [review]

File: .github/workflows/storybook-tests.yml [hunk 1]
```diff
@@ -0,0 +1,24 @@
+# .github/workflows/storybook-tests.yml
+
+name: 'Storybook Tests'
+on: push
```
[the remark this answers is not in the document]
Comment (Gkrumbach07) [reply .github/workflows/storybook-tests.yml:2]: i can change it to on pr. makes more sense

File: .github/workflows/storybook-tests.yml [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+jobs:
+  test:
+    timeout-minutes: 60
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
```
[the remark this answers is not in the document]
Comment (Gkrumbach07) [reply .github/workflows/storybook-tests.yml:8]: reason: this is what storybooks docs gave as an example. Im sure v3 would be fine. will make the switch

File: .github/workflows/storybook-tests.yml [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+      - uses: actions/setup-node@v2
+        with:
+          node-version: '18.x'
```
[the remark this answers is not in the document]
Comment (Gkrumbach07) [reply .github/workflows/storybook-tests.yml:11]: do we need that now or is keeping it at 18 fine?

**Comment by Gkrumbach07** [comment]
command is failing in the github action... investigating

**Comment by Gkrumbach07** [comment]
everything is caching properly now. and looks like it is working via the checkmark in the actions list

File: .github/workflows/storybook-tests.yml [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+      - name: Install Node.js packages
+        working-directory: ./frontend
+        if: ${{ steps.frontend-cache.outputs.cache-hit != 'true' }}
+        run: npm install
+      - name: Install dependencies
+        working-directory: ./frontend
+        run: npm install
+      - name: Install Playwright
+        working-directory: ./frontend
+        run: npx playwright install --with-deps
```
Comment (andrewballantyne) [inline .github/workflows/storybook-tests.yml:33]: Is there a reason this is not covered as part of the install dependencies?

**Review by andrewballantyne — CHANGES_REQUESTED** [review]

File: frontend/src/pages/projects/screens/projects/ProjectView.stories.tsx [hunk 5]
```diff
@@ -135,8 +135,8 @@ CreateProject.play = async ({ canvasElement }) => {
   // get modal
   const body = within(canvasElement.ownerDocument.body);
   body.getByLabelText('Name');
-  const resourceNameInput = body.getByRole('textbox', { name: /Resource name/ });
-  const nameInput = body.getByRole('textbox', { name: /Name/ });
+  const resourceNameInput = body.getAllByRole('textbox').at(1) as HTMLElement;
+  const nameInput = body.getByRole('textbox', { name: 'Name' });
```
[the remark this answers is not in the document]
Comment (Gkrumbach07) [reply frontend/src/pages/projects/screens/projects/ProjectView.stories.tsx:139]: hmm i must have done a merge without looking. my bad. i will revert that chunk back

**Pull request merged**